setup.py

```diff
@@ -106,6 +106,8 @@
     WITH_DISTRIBUTED_MW, WITH_GLOO_IBVERBS
 
 DEBUG = check_env_flag('DEBUG')
+ENABLE_IBVERBS_ENV = check_env_flag('WITH_GLOO_IBVERBS')
+WITH_GLOO_IBVERBS = WITH_GLOO_IBVERBS and ENABLE_IBVERBS_ENV
```

Comment:
I think the logic for handling these should go into `tools/setup_helpers/dist_check.py`